### PR TITLE
dApp: Transfer screen redirect when no channels exist

### LINKED ISSUE #2039: Transfer screen is shown after the last channel has been settled
## Steps to Reproduce
1. Settle the last channel
2. Close the browser window
3. Connect to the dApp

## Expected Result
- Observe the transfer screen
- Dialog "You don't have channels any more" -> "Continue" button
- Afterward the "Start" screen is shown

    grafik

## Actual Result
- The transfer screen is shown
- If you click on "Channel" there is no channel shown
- If you click on "Change Token" there is no connected token shown

    grafik

## Additional Infos (Optional) 
```
- Browser & version
- Browser console errors
- Device / OS
```

### PR DESCRIPTION
**Pull request by taleldayekh**
Fixes #2039

**Short description**
Whenever all channels for all connected tokens are closed and settled a dialog appears warning the user about not having any connected open channels or connected tokens. The dialog has a button that redirects the user to the token select screen.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Connect to a token and open a channel.
2. Close and settle the channel.

### CODE REVIEW

File: raiden-dapp/src/components/dialogs/NoChannelsDialog.vue [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import RaidenDialog from '@/components/dialogs/RaidenDialog.vue';
+import ActionButton from '@/components/ActionButton.vue';
+import NavigationMixin from '../../mixins/navigation-mixin';
+
+@Component({
+  components: { RaidenDialog, ActionButton },
+})
+export default class NoChannelsDialog extends Mixins(NavigationMixin) {
+  @Prop({ required: true })
+  visible!: boolean;
```
Comment (weilbith) [inline raiden-dapp/src/components/dialogs/NoChannelsDialog.vue:31]: Maybe I lack in knowledge about the dialog, but I would rather put the render condition where used in the parent component and not here with a prop. Else I think we could end up in an awkward  mix of different pattern.

**Review by weilbith — APPROVED** [review]
Really good!
I like that we discussed together first to find the best solution. Well done.

Unfortunately I can't test it right now. And I also miss details on Vuetify. So can the dialog be closed else than clicking on the action button? Because that should not be possible. Optimistic approval that this works. :)

File: raiden-dapp/src/components/dialogs/NoChannelsDialog.vue (hunk 1 above)
Comment (weilbith) [reply raiden-dapp/src/components/dialogs/NoChannelsDialog.vue:31]: Ignore this comment. I think this is perfectly fine.

**Comment by taleldayekh** [comment]
> So can the dialog be closed else than clicking on the action button?

Usually, you can also close the dialog by clicking outside it but when the `hide-close` prop is set you can only close it by manually emitting a close event.

**Comment by taleldayekh** [comment]
Thanks for taking the time to review @weilbith 👍

**Pull request merged**